### PR TITLE
Update react to 18 in package.json template

### PR DESCRIPTION
**Pull request by academo**
**What this PR does / why we need it**:

Updates the template react and react-dom to the same as grafana 10.0.x

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @grafana/create-plugin@2.9.1-canary.455.5613aaf.0
  npm install @grafana/sign-plugin@2.2.1-canary.455.5613aaf.0
  # or 
  yarn add @grafana/create-plugin@2.9.1-canary.455.5613aaf.0
  yarn add @grafana/sign-plugin@2.2.1-canary.455.5613aaf.0
  ```
</details>

### CODE REVIEW

**Review by jackw — APPROVED** [review]
LGTM! 🚀 

Any reason for pinning the versions of `emotion`, `react-router` and `testing-library`?

**Comment by academo** [comment]
> LGTM! 🚀
> 
> Any reason for pinning the versions of `emotion`, `react-router` and `testing-library`?

@jackw not in particular. should i keep the hat (`^`) there?

**Comment by jackw** [comment]
> not in particular. should i keep the hat (^) there?

I think we should keep the caret in there otherwise on scaffold they get the specific version rather than the latest of that major.

File: packages/create-plugin/templates/common/_package.json [hunk 1]
```diff
@@ -26,8 +26,9 @@
     "@swc/core": "^1.3.90",
     "@swc/helpers": "^0.5.0",
     "@swc/jest": "^0.2.26",
-    "@testing-library/jest-dom": "^5.16.5",
-    "@testing-library/react": "^12.1.4",
+    "@testing-library/jest-dom": "6.1.4",
+    "@types/testing-library__jest-dom": "5.14.8",
```
Comment (jackw) [inline packages/create-plugin/templates/common/_package.json:30]: mega nit: move this to line 36 so package managers don't move it "randomly" when someone adds a package via cli.

**Review by jackw — APPROVED** [review]
LGTM! 🚀

**Pull request merged**